packages/textfield/src/openfl/text/TextField.hx

```diff
@@ -1376,11 +1376,20 @@ class TextField extends InteractiveObject
 
 		__updateScrollV();
 		__updateScrollH();
+		
+		#if dom 
+		if(stage!=null && stage.focus == this && __renderedOnCanvasWhileOnDOM)
+		{
+			__stopCursorTimer();
+			__startCursorTimer();
+		}
+		#else
 		if(stage!=null && stage.focus == this)
```

Comment:
This is creating duplicate code, I would suggest in-lining the if dom to prevent code duplication.

if(stage!=null && stage.focus == this 
#if dom && __renderedOnCanvasWhileOnDOM #end)

Something like that.

Reply:
Could also be something like

var domCaretPositionFix = #if dom __renderedOnCanvasWhileOnDOM #else true #end